packages/tools/examples/segmentSelect/index.ts

```diff
@@ -0,0 +1,303 @@
+      '1.3.6.1.4.1.14519.5.2.1.7009.2403.367700692008930469189923116409',
+    wadoRsRoot: 'https://d33do7qe4w26qo.cloudfront.net/dicomweb',
+  });
+
+  // Instantiate a rendering engine
+  renderingEngine = new RenderingEngine(renderingEngineId);
+
+  // Create the viewports
+  const viewportInputArray = [
+    {
+      viewportId: stackViewportId,
+      type: ViewportType.STACK,
+      element: element1,
+    },
+    {
+      viewportId: volumeViewportId,
+      type: ViewportType.ORTHOGRAPHIC,
+      element: element2,
+      defaultOptions: {
+        orientation: Enums.OrientationAxis.SAGITTAL,
+      },
+    },
+  ];
+  renderingEngine.setViewports(viewportInputArray);
+
+  stackToolGroup.addViewport(stackViewportId, renderingEngineId);
+  volumeToolGroup.addViewport(volumeViewportId, renderingEngineId);
+
+  _handleStackViewport(stackImageIds);
+  _handleVolumeViewport(volumeImageIds, renderingEngine);
+}
+
+run();
+
+async function _handleVolumeViewport(volumeImageIds, renderingEngine) {
+  const volume = await cornerstone.volumeLoader.createAndCacheVolume(volumeId, {
+    imageIds: volumeImageIds,
+  });
+
+  // Set the volume to load
+  volume.load();
+
+  // Set volumes on the viewports
+  await cornerstone.setVolumesForViewports(
+    renderingEngine,
+    [{ volumeId, callback: setCtTransferFunctionForVolumeActor }],
+    [volumeViewportId]
+  );
+
+  // Create a segmentation of the same resolution as the source data
+  await cornerstone.volumeLoader.createAndCacheDerivedSegmentationVolume(
+    volumeId,
+    {
+      volumeId: volumeSegmentationId,
+    }
+  );
+
+  createMockEllipsoidVolumeSegmentation({
+    volumeId: volumeSegmentationId,
+    cornerstone,
+  });
+
+  // Add the segmentations to state
+  segmentation.addSegmentations([
+    {
+      segmentationId: volumeSegmentationId,
+      representation: {
+        // The type of segmentation
+        type: csToolsEnums.SegmentationRepresentations.Labelmap,
+        // The actual segmentation data, in the case of labelmap this is a
+        // reference to the source volume of the segmentation.
+        data: {
+          volumeId: volumeSegmentationId,
+        },
+      },
+    },
+  ]);
+
+  // Add the segmentation representation to the toolgroup
+  segmentation.addSegmentationRepresentations(volumeToolGroupId, [
+    {
+      segmentationId: volumeSegmentationId,
+      type: csToolsEnums.SegmentationRepresentations.Labelmap,
+    },
+  ]);
+}
+
+async function _handleStackViewport(stackImageIds: string[]) {
+  const stackViewport = renderingEngine.getViewport(stackViewportId);
+  const imageIdsArray = [stackImageIds[0]];
+
+  const { imageIds: segmentationImageIds } =
+    await imageLoader.createAndCacheDerivedSegmentationImages(imageIdsArray);
+
+  await stackViewport.setStack(imageIdsArray, 0);
+
+  createMockEllipsoidStackSegmentation({
+    imageIds: imageIdsArray,
+    segmentationImageIds,
+    cornerstone,
+  });
+
+  segmentation.addSegmentations([
```

Comment:
We are starting to get an awful lot of this boilerplate code that really is probably shared.  What about having a few more helper methods to setup some of these sorts of things for the examples?

Reply:
I think it is fine for examples, honestly. It is a lot of work. Examples don't hurt to be explicit about the APIs. The mock was a utility rather than showing how our API works.

Reply:
I refactored more mocking data instead